### PR TITLE
 Add Twilio subscription via websub

### PR DESCRIPTION
**Pull request by Chuhaa**
## Purpose
Add Twilio subscription via websub

## Goals
Resolves https://github.com/wso2-enterprise/choreo/issues/14190

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Test environment
2201.2.1

### CODE REVIEW

File: asyncapi/twilio/data_types.bal [hunk 1]
```diff
@@ -14,6 +14,24 @@
...
+# + authToken - The authentication token of the account 
+# + phoneNumberSid - The Twilio-provided string that uniquely identifies the phone number
+# + callbackURL - Callback service URL for webhook
+@display {label: "Listener Config"}
+public type ListenerConfig record {
+    @display {label: "Account SID", "description": "Twilio account SID"}
+    string accountSId;
+    @display {label: "Auth Token", "description": "The authentication token of the account"}
+    string authToken;
+    @display {label: "Phone Number Sid", "description": "The Twilio-provided string that uniquely identifies the phone number"}
```
Comment (sachinira) [inline asyncapi/twilio/data_types.bal:29]: Can we make this `SID`?

File: asyncapi/twilio/data_types.bal [hunk 2]
```diff
@@ -14,6 +14,24 @@
...
+    string phoneNumberSid;
+    @display {label: "Call Back URL", "description": "Callback service URL for webhook"}
```
Comment (sachinira) [inline asyncapi/twilio/data_types.bal:31]: Can we make Call Back one word?

File: asyncapi/twilio/listener.bal [hunk 3]
```diff
@@ -15,23 +15,32 @@
 // under the License.
 
 import ballerina/http;
+import ballerina/log;
+import ballerina/url;
+import ballerina/mime;
```
Comment (kasthuriraajan) [inline asyncapi/twilio/listener.bal:20]: Shall we organize the imports in alphabetic order ?

**Review by kasthuriraajan — DISMISSED** [review]

File: asyncapi/twilio/listener.bal (hunk 3 above)
Comment (Chuhaa) [reply asyncapi/twilio/listener.bal:19]: Fixed in https://github.com/ballerina-platform/asyncapi-triggers/commit/475116df52f839b12941b4cdcc5d908a08f382f0

File: asyncapi/twilio/data_types.bal (hunk 2 above)
Comment (Chuhaa) [reply asyncapi/twilio/data_types.bal]: Fixed in https://github.com/ballerina-platform/asyncapi-triggers/commit/475116df52f839b12941b4cdcc5d908a08f382f0

File: asyncapi/twilio/data_types.bal (hunk 1 above)
Comment (Chuhaa) [reply asyncapi/twilio/data_types.bal]: Fixed in https://github.com/ballerina-platform/asyncapi-triggers/commit/475116df52f839b12941b4cdcc5d908a08f382f0

**Comment by Drifftr** [comment]
Since this involves an API change we have to update related samples with this PR. And need to do a major version bump with it for the trigger.

**Comment by NipunaRanasinghe** [comment]
Closing as per https://github.com/wso2-enterprise/choreo/issues/14190#issuecomment-2329032315

**Pull request closed without merging**